src/providers/user-data.ts

```diff
@@ -1,11 +1,54 @@
 import { Injectable } from '@angular/core';
+import { Response, Http } from '@angular/http';
+import { Storage } from '@ionic/storage';
+import { AuthHttp, tokenNotExpired } from 'angular2-jwt';
+import { StockpileData } from './stockpile-data';
+import { Links } from '../constants';
+import 'rxjs/add/operator/map';
 
 @Injectable()
 export class UserData {
+  storage = new Storage();
 
-  constructor() { }
+  constructor(
+    public authHttp: AuthHttp,
+    public stockpileData: StockpileData,
+    public http: Http
+  ) { }
 
   login(email: string, password: string) {
-    return Promise.resolve();
+    let creds = {
+      'email': email,
+      'password': password
+    };
+
+    return new Promise((resolve, reject) => {
+      this.http.post(this.stockpileData.getUrl(Links.authenticate), creds)
```

Comment:
I'm actually not sure I'm using HAL the right way here. There is no hard coded urls, but we need hard coded strings to get the urls (they are in `constants.ts`). It does seem like a slight improvement (urls can be changed more easily), but still feels odd to do this.